moped-editor/src/views/projects/projectView/NoteInput.js

```diff
@@ -88,7 +102,8 @@ const OnChangePlugin = ({ onChange }) => {
   useEffect(() => {
     return editor.registerUpdateListener(({ editorState }) => {
       editorState.read(() => {
-        const htmlContent = $generateHtmlFromNodes(editor, null);
+        const isEditorEmpty = $isRootTextContentEmpty();
+        const htmlContent = isEditorEmpty ? null : $generateHtmlFromNodes(editor, null);
         onChange(htmlContent);
       });
     });
```

Comment:
Could you add a dependency array to this useEffect and put `editor` in it like in the other plugin components in this file? Otherwise, this code will run on every render like it wasn't wrapped in useEffect at all.

Reply:
thanks for catching that!